lib/Jsrt/ChakraCore.h

```diff
@@ -163,4 +163,346 @@ JsGetModuleHostInfo(
+///     Parses a script and returns a function representing the script.
+/// </summary>
+/// <remarks>
+///     Requires an active script context.
+/// </remarks>
+/// <param name="script">The script to run.</param>
+/// <param name="sourceContext">
+///     A cookie identifying the script that can be used by debuggable script contexts.
+/// </param>
+/// <param name="sourceUrl">The location the script came from.</param>
+/// <param name="parseAttributes">Attribute mask for parsing the script</param>
+/// <param name="result">The result of the compiled script.</param>
+/// <returns>
+///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+/// </returns>
+CHAKRA_API
+    JsCompile(
+        JsValueRef script,
+        JsSourceContext sourceContext,
+        JsValueRef sourceUrl,
+        JsParseScriptAttributes parseAttributes,
+        _Out_ JsValueRef *result);
+
+/// <summary>
+///     Executes a script.
+/// </summary>
+/// <remarks>
+///     Requires an active script context.
+/// </remarks>
+/// <param name="script">The script to run.</param>
+/// <param name="sourceContext">
+///     A cookie identifying the script that can be used by debuggable script contexts.
+/// </param>
+/// <param name="sourceUrl">The location the script came from</param>
+/// <param name="parseAttributes">Attribute mask for parsing the script</param>
+/// <param name="result">The result of the script, if any. This parameter can be null.</param>
+/// <returns>
+///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+/// </returns>
+CHAKRA_API
+    JsRun(
+        JsValueRef script,
+        JsSourceContext sourceContext,
+        JsValueRef sourceUrl,
+        JsParseScriptAttributes parseAttributes,
+        _Out_ JsValueRef *result);
+
+/// <summary>
+///     Gets the property ID associated with the name.
+/// </summary>
+/// <remarks>
+///     <para>
+///     Property IDs are specific to a context and cannot be used across contexts.
+///     </para>
+///     <para>
+///     Requires an active script context.
+///     </para>
+/// </remarks>
+/// <param name="name">
+///     The name of the property ID to get or create. The name may consist of only digits.
+///     The string is expected to be encoded as utf8.
+/// </param>
+/// <param name="length">length of the name</param>
+/// <param name="propertyId">The property ID in this runtime for the given name.</param>
+/// <returns>
+///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+/// </returns>
+CHAKRA_API
+    JsCreatePropertyIdUtf8(
+        _In_z_ const char *name,
+        size_t length,
+        _Out_ JsPropertyIdRef *propertyId);
+
+/// <summary>
+///     Gets the name associated with the property ID.
+/// </summary>
+/// <remarks>
+///     <para>
+///     Requires an active script context.
+///     </para>
+///     <para>
+///     Consumer is responsible from calling JsStringFree to free allocated memory.
```

Comment:
Is there still `JsStringFree`? Correct me if I'm wrong, I thought ChakraCore copies propertyid string and manages its lifetime.